src/calculateAverage.js

```diff
@@ -37,7 +37,35 @@ function applyCalculateAverage() {
    * @param {Object} options - optional
    */
   [].__proto__.calculateAverage = function(options) {
-    // write code here
+    if (options === null) {
+      return undefined;
+    } else if (options === undefined) {
+      return this.reduce((acc, elem) => acc + elem, 0) / this.length;
+    }
+
+    if (options.hasOwnProperty('propertyName')
+      && this.some(elem => !!elem[options.propertyName]
+        && typeof elem[options.propertyName] === 'number')) {
```

Comment:
Do you really need it if you have if inside reduce? Will be code without this check faster or slower?

Reply:
yes, I need it in order to avoid NaN as result of my reduce